### ISSUE TITLE
Does not generate files?

### ISSUE THREAD

**Issue by ignatandrei** [farskeptic/AutoCompose#1]
I have tried your project to add to my list of RSCG

https://ignatandrei.github.io/RSCG_Examples/v2/docs/List-of-RSCG

However, when creating a sample project does not generate files 

When compiling , it says: 
1>CSC : warning CS8785: Generator 'AutoComposeSourceGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'ArgumentNullException' with message 'Value cannot be null.'.

attached the project to can reproduce the issue

[Decorator.zip](https://github.com/farskeptic/AutoCompose/files/14472270/Decorator.zip)

**Comment by farskeptic** [comment]
Hi @ignatandrei - Thanks for raising this.  

The issues that are causing your failure are as follows:
1. Because this is source generation at compile time, nameof(x) does not resolve to a string.  I'll have to look for either a literal or a nameof.
2. The autogenerated code assumes class and interface are public.  I'll have to detect what accessor the class is using, and use the same.
3. The exception is not informative enough.

As a workaround, if you make the following modifications to your project, it will work:
1. Make your class Coffee public
4. Make your interface ICoffee public
5. Change nameof(_cof) to "_cof"

Thanks again for raising this.  I'll make the fixes and push a new version as soon as I'm able.



So:

```csharp

public interface ICoffee
{
    :
}
```

```csharp
[AutoCompose(typeof(ICoffee), "_cof")]
public partial class CoffeeWithLogging : ICoffee
{
    protected ICoffee _cof;

    public CoffeeWithLogging(ICoffee cof)
    {
        _cof = cof;
    }
}
```

**Comment by farskeptic** [comment]
Hi @ignatandrei - this has been fixed in 1.0.1.  You can use your original project upgraded to v1.0.1 of the package and it will work.  Please let me know if you have any further issues.

**Comment by ignatandrei** [comment]
Done implementing at https://ignatandrei.github.io/RSCG_Examples/v2/docs/Farskeptic.AutoCompose

Thanks!
Andrei
